tianshou/data/batch.py

```diff
@@ -500,6 +503,35 @@ def __getitem__(self, index: str | IndexType) -> Any:
             return new_batch
         raise IndexError("Cannot access item from empty Batch object.")
 
+    def __eq__(self, other: Self) -> bool:  # type: ignore
+        this_dict = self.__dict__
```

Comment:
See my comment above - without the isinstance this can give equality between different object that just both happen to have the same `__dict__`. Moreover, `==` with a non-batch object can cause an attribute error.